### PR TITLE
support multiple files

### PR DESCRIPTION
**Pull request by amir-arad**
I have several test files, it helps me organise the API regression tests.
This feature can really ease things for me

### CODE REVIEW

**Comment by danvk** [comment]
Hi @amir-arad! Have you run across dtslint? It serves a similar purpose as typings-checker but is actively maintained. Does it work for you?

**Comment by amir-arad** [comment]
@danvk first I've heard of dtslint. looks promising, thanks!
I came across typings-checker through @gcanti 's typelevel-ts project. 

May I suggest a notice in the readme file that directs potential users to dtslint?

**Comment by amir-arad** [comment]
@danvk I've tried out dtslint, it's not for my use case.

I'm using typings-checker to thoroughly test the negative and positive signatures of the API of my typescript library. 
dtslint assumes there is a `index.d.ts` file to tests, installs its own typescript and a whole bunch of weird stuff. It's explicitly isnt designed for typescript library authors.

If you don't feel like maintaining this library, I will be glad to take it.

File: src/index.ts [hunk 1]
```diff
@@ -35,48 +35,49 @@ const options: ts.CompilerOptions = ts.readConfigFile(project || 'tsconfig.json'
...
+        ts.ScriptTarget.ES5, false, ts.LanguageVariant.Standard, source.getFullText());
 
-const source = program.getSourceFile(tsFile);
-if (!source) {
-  console.error(`could not load content of ${tsFile}`);
-  process.exit(1);
-}
-const scanner = ts.createScanner(
-    ts.ScriptTarget.ES5, false, ts.LanguageVariant.Standard, source.getFullText());
+    const checker = program.getTypeChecker();
```
Comment (danvk) [inline src/index.ts]: Can this go outside the forEach loop?

File: src/index.ts [hunk 2]
```diff
@@ -35,48 +35,49 @@ const options: ts.CompilerOptions = ts.readConfigFile(project || 'tsconfig.json'
     .config.compilerOptions || {};
 const host = ts.createCompilerHost(options, true);
 
-const program = ts.createProgram([tsFile], options, host);
+const program = ts.createProgram(tsFiles, options, host);
+let totalNumFailures = 0;
+tsFiles.forEach((tsFile: string) => {
+    const source = program.getSourceFile(tsFile);
```
Comment (danvk) [inline src/index.ts]: The rest of the code uses a 2-space indent, so stick with that.

Comment (amir-arad) [inline src/index.ts]: done

File: src/index.ts (hunk 1 above)
Comment (amir-arad) [inline src/index.ts]: I just found out that I can't even re-use the program for all the files. working on it, but for now I've committed a very dumb solution.

**Comment by danvk** [comment]
@amir-arad do you want to merge this?

**Comment by amir-arad** [comment]
@danvk I was hoping to come up with a better solution than this, but It's been too long and I guess it's just an optimization.

**Comment by amir-arad** [comment]
@danvk my npm profile is https://www.npmjs.com/~amir-arad 
Thanks

**Comment by amir-arad** [comment]
@danvk waiting for permissions in [npm](https://www.npmjs.com/package/typings-checker) to publish this

**Comment by danvk** [comment]
@amir-arad Added you!

**Pull request merged**